src/components/App/App.js

```diff
@@ -18,7 +25,93 @@ function App() {
   const [isSignupPopupOpen, setIsSignupPopupOpen] =
     React.useState(false);
   const [isRegistrationSuccessfulOpen, setIsRegistrationSuccessfulOpen] = React.useState(false);
-  const [isLoggedIn, setIsLoggedIn] = React.useState(false);// eslint-disable-line
+  const [isLoggedIn, setIsLoggedIn] = React.useState(false);
+  const [name, setname] = React.useState("");
+  const [currentUser, setCurrentUser] = React.useState({});
+  const [searchedArticles, setSearchedArticles] = React.useState((JSON.parse(localStorage.getItem("searchResults"))));
+  const [isSeacrhProcessing, setIsSearchProcessing] = React.useState(false);
+  const [submitSearch, setSubmitSearch] = React.useState(false);
+  const [keyword, setKeyword] = React.useState(localStorage.getItem("keyword"));
+  const navigate = useNavigate();
+  const [savedArticles, setSavedArticles] = React.useState([]);
+
+
+  function register(email, password, name) {
+    auth.signup(email, password, name)
+      .then((res) => {
+        console.log(res._id)
+      })
+      .then(() => {
+        closeAllPopups();
+        setIsRegistrationSuccessfulOpen(true);
+      })
+      .catch((err) => {
+        console.log(`Error: ${err.status}`);
+      })
+  }
+
+  function login(email, password) {
+    auth.signin(email, password)
+      .then(res => {
+        if (res.token) {
+          setIsLoggedIn(true);
+          setname(res.name);
+          localStorage.setItem("jwt", res.token);
+          navigate("/");
+        } else {
+          console.log("failed to sign-in");
+        }
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => closeAllPopups());
```

Comment:
`      .finally(() => closeAllPopups());
`

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

Reply:
Done